### PR TITLE
Fix mutter 3.3x crashes

### PR DESCRIPTION
**Pull request by Tireg**
This PR fixes issues against running upstream gala with mutter 3.3x.
This should fix https://github.com/elementary/gala/issues/634

Every single patch is detailed in this post: https://github.com/elementary/gala/issues/634#issuecomment-551851873
I reported every step as a commit comment in order to understand it more easily.

### CODE REVIEW

File: src/Widgets/MultitaskingView.vala [hunk 1]
```diff
@@ -389,29 +392,20 @@ namespace Gala
 			// FIXME is there a better way to get the removed workspace?
 #if HAS_MUTTER330
             unowned Meta.WorkspaceManager manager = display.get_workspace_manager ();
+			unowned List<Workspace> existing_workspaces = null;
```
Comment (tintou) [inline src/Widgets/MultitaskingView.vala]:
```suggestion
			List<Workspace> existing_workspaces = null;
```

File: src/WorkspaceManager.vala [hunk 2]
```diff
@@ -132,10 +132,16 @@ namespace Gala
 
 		void workspace_removed (Meta.WorkspaceManager manager, int index)
 		{
+			unowned List<Workspace> existing_workspaces = null;
```
Comment (tintou) [inline src/WorkspaceManager.vala]:
```suggestion
			List<Workspace> existing_workspaces = null;
```

File: src/WorkspaceManager.vala [hunk 3]
```diff
@@ -389,9 +395,13 @@ namespace Gala
 #if HAS_MUTTER330
             unowned Meta.Display display = wm.get_display ();
 			unowned Meta.WorkspaceManager manager = display.get_workspace_manager ();
-			var last_index = manager.get_n_workspaces () - 1;
+			unowned List<Meta.Workspace> workspaces = null;
```
Comment (tintou) [inline src/WorkspaceManager.vala]:
```suggestion
			List<Meta.Workspace> workspaces = null;
```

**Comment by Tireg** [comment]
> By creating an `unowned List` you're creating a leak here

For each one of the suggestions, we fill the list using `manager.get_workspace_by_index (i)`. I don't know Vala but isn't there something like `List<unowned Workspace>` (instead of `List<Workspace>`) that I should use in order to avoid destroying the `Workspace`s objects when the `List` is destroyed ?

Just to be sure there is no mistake.

**Comment by tintou** [comment]
@Tireg a new commit is fine too, no worries :)
The difference here:
`List<unowned Workspace>` means that we have to free the List but not the elements
`List<Workspace>` means that we have to free the list and unref the elements
Defined as `List<Workspace>`, Vala will take care to increase the reference count before adding an element to the list, when `List<unowned Workspace>` would not ref the workspace, so `List<Workspace>` is safer for us in this case.

**Comment by decathorpe** [comment]
I'm looking forward to testing this on fedora :)

Just one question / nit-pick: The name for the gsettings override file seems rather odd:

`20_elementary.pantheon.wm.gschema.override`

Can we rename it to something like `20_io.elementary.gala.gschema.override` instead?

**Comment by tintou** [comment]
@decathorpe Yeah, would make sense too :)

**Comment by decathorpe** [comment]
Hm. I'm still getting a reproducible crash on startup on fedora 31 :( I'll try to file a new bug report for it, once I'm able to provide a useful traceback :(

**Pull request merged**